### PR TITLE
BD now works even when BD not defined in mid chord

### PR DESCRIPTION
**Pull request by ptrbortolotti**

### CODE REVIEW

File: pyFAST/converters/beamDynToHawc2.py [hunk 1]
```diff
@@ -87,14 +88,60 @@ def beamDynToHawc2(BD_mainfile, BD_bladefile, H2_htcfile=None, H2_stfile=None, b
...
             K[i,j]=prop['K{}{}'.format(i+1,j+1)].values
             M[i,j]=prop['M{}{}'.format(i+1,j+1)].values
 
+    if fstIn != None:
+        fst = FASTInputDeck(fstIn, verbose=True)
+        Bld = fst.fst_vt['AeroDynBlade']
+        AF = fst.fst_vt['af_data']
+        BlSpn = Bld['BldAeroNodes'][:,0]
+        n_span = len(BlSpn)
+        ac_pos = np.zeros(n_span)
```
Comment (ebranlard) [inline pyFAST/converters/beamDynToHawc2.py:105]: Could this be renamed somthing like 
`le2ac_raw = np.zeros(n_span) # Leading edge to aerodynamic center (in chord)`

File: pyFAST/converters/beamDynToHawc2.py [hunk 2]
```diff
@@ -87,14 +88,60 @@ def beamDynToHawc2(BD_mainfile, BD_bladefile, H2_htcfile=None, H2_stfile=None, b
...
+        for iSpan in range(n_span):
+            ac_pos[iSpan] = fst.fst_vt['ac_data'][iSpan]['AirfoilRefPoint'][0]
+        # Define axis of aero centers
+        BlCrvAC = Bld['BldAeroNodes'][:,1]
+        BlSwpAC = Bld['BldAeroNodes'][:,2]
+        chord = Bld['BldAeroNodes'][:,5]
+        s_aero = BlSpn/BlSpn[-1]
+        ac_x = np.interp(r_bar, s_aero, BlCrvAC)
+        ac_y = np.interp(r_bar, s_aero, BlSwpAC)
+        ac = np.interp(r_bar, s_aero, ac_pos)
```
Comment (ebranlard) [inline pyFAST/converters/beamDynToHawc2.py:115]: Could this be renamed something like
le2ac = np.interp(r_bar, s_aero, le2ac_raw)

File: pyFAST/converters/beamDynToHawc2.py [hunk 3]
```diff
@@ -87,14 +88,60 @@ def beamDynToHawc2(BD_mainfile, BD_bladefile, H2_htcfile=None, H2_stfile=None, b
...
+
+        # Get x and y coordinates of c2 axis
+        ac2c2 = (0.5 - ac) * chord
+        c2_x = ac_x + ac2c2 * np.sin(twist)
+        c2_y = ac_y + ac2c2 * np.cos(twist)
+        # Get offsets from BD axis to c2 axis along the twisted frame of reference
+        c2BD_y = (c2_y - kp_y) / np.cos(twist)
```
Comment (ebranlard) [inline pyFAST/converters/beamDynToHawc2.py:122]: I might be wrong, but maybe:
`c2BD_y = np.sqrt(  (c2_y - kp_y)**2  + (c2_xy - kp_x)**2  )`
is more robust?  Please double check!

**Review by ebranlard — CHANGES_REQUESTED** [review]
Thanks a lot for doing that, I have some small suggestions

File: pyFAST/converters/beamDynToHawc2.py (hunk 3 above)
Comment (ptrbortolotti) [reply pyFAST/converters/beamDynToHawc2.py]: done, I don't think it changes anything

File: pyFAST/converters/beamDynToHawc2.py (hunk 2 above)
Comment (ptrbortolotti) [reply pyFAST/converters/beamDynToHawc2.py]: done

File: pyFAST/converters/beamDynToHawc2.py (hunk 1 above)
Comment (ptrbortolotti) [reply pyFAST/converters/beamDynToHawc2.py]: done, good idea

**Review by ebranlard — APPROVED** [review]

**Pull request merged**